### PR TITLE
ndjson writer

### PR DESCRIPTION
**Pull request by mattnibs**
Support writing output to native ndjson.

### CODE REVIEW

File: test/btest.cfg [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [btest]
-TestDirs    = procs
+TestDirs    = procs cmd
```
Comment (nwt) [inline test/btest.cfg]: Nit: Maybe keep these sorted.

File: pkg/zsio/ndjson/writer.go [hunk 2]
```diff
@@ -10,22 +10,17 @@ import (
...
+	encoder *json.Encoder
 }
 
 func NewWriter(w io.WriteCloser) *NDJSON {
-	return &NDJSON{Writer: zson.Writer{w}}
+	writer := &NDJSON{
+		Writer: zson.Writer{w},
+	}
+	writer.encoder = json.NewEncoder(writer.Writer)
+	return writer
```
Comment (nwt) [inline pkg/zsio/ndjson/writer.go]: Nit: I think you can simplify this to
```go
	return &NDJSON{
		Writer: zson.Writer{w},
		encoder: json.NewEncoder(w),
	}
```

File: pkg/zson/record.go [hunk 3]
```diff
@@ -293,6 +294,15 @@ func (r *Record) CutTypes(fields []string) ([]zeek.Type, uint64) {
 	return nil, found
 }
 
+// MarshalJSON implements json.Mashaler.
```
Comment (nwt) [inline pkg/zson/record.go]: s/Mashaler/Marshaler/

File: test/btest.cfg (hunk 1 above)
Comment (nwt) [reply test/btest.cfg]: Separately, I think format/ or output/ or flag/f/ would be a better name for the test directory.

File: pkg/zsio/ndjson/writer.go (hunk 2 above)
Comment (mccanne) [reply pkg/zsio/ndjson/writer.go]: I would call this Writer instead of NDJSON (my bad when I created the stub), i.e., ndjson.Writer

**Comment by mccanne** [comment]
Hmm, seeing all the MarshalJSON methods makes me think all the scalar types should not be structs, e.g., 
```
type Int int64
```
instead of 
```
type Int struct {
    Native int64
}
```
Then the default Marshal code would work without the marshaler (right?)
Maybe we could fix this in a subsequent PR.

File: pkg/zsio/ndjson/writer.go (hunk 2 above)
Comment (mattnibs) [reply pkg/zsio/ndjson/writer.go]: Oh yeah, duh

**Pull request merged**